### ISSUE TITLE
RFC: A tz aware Timex.today

### ISSUE THREAD

**Issue by rvignesh89** [bitwalker/timex#300]
In my use case I have to deal with dates which are dependant on time zone. It would be nice to have something like

```
Timex.today("Asia/Singapore")
```
Currently I do this to achieve the same which is a slightly verbose IMO,

```
Timex.to_date(Timex.now("Asia/Singapore"))
```

A minor improvement but enhances the readability of the code a lot. Also compares nicely with the Timex.now syntax.

**Comment by bitwalker** [comment]
I'm hesitant to do this, and I'll try to explain why, hopefully people can chime in with their two cents here:

I want to try and avoid conflating the ideas of timezone-less structures with timezone-aware concepts (for example, asking for a Date in a given timezone). Yes, it is a natural question one might ask in a conversation, but the concept of the `Date` struct in terms of a timezone is nonsensical from a technical point of view. You *must* have a full `DateTime` in order to reason about what day it is in some timezone (and in fact `Timex.today/0` effectively does just this to give you what day it is right now in UTC). On top of this, looking back through history, and depending on the timezone, there are times where on the same day, two distinctly different zones were in effect on the same day. This means that ultimately, outside of UTC, you can't reason about what day it is unless you specify a time. `Timex.today` itself should probably not exist, as I doubt it's particularly helpful anyway, but it's an artifact from a time where I tried to blur the lines between `Date` and `DateTime` to make working with them more natural. Unfortunately dates, times, and timezones are not intuitive at all, and at times contradictory and nonsensical, and trying to make them appear sane and intuitive can actually cause more problems than not.

So, TL;DR, I prefer to not make `Date`-specific functions timezone aware. If you are writing code like in your example, you should just call `Timex.now("Asia/Singapore")` and use the full `DateTime` object - it doesn't make sense to drop the time when it's clearly needed to establish what day it actually is in that timezone. All of the Timex APIs will take any of the calendar structs and operate on them without conversion, so you shouldn't have to modify any of your code which works with the resulting value.

**Comment by rvignesh89** [comment]
@bitwalker I was caught up with other stuff so couldn't reply to this earlier. I agree with you on the fact that dates, times and timezones are not intuitive and require some mind wrapping to get it it right especially if you application does not have a global timezone. This is exactly why I wanted `Timex.today('Asia/Singapore')` since it would help in making the code explain what I'm trying to do succinctly.

However I understand your reason to not go down this path. Thanks for your explanation! I'll leave the issue open for others to chime in their thoughts.

**Comment by bitwalker** [comment]
I'm going to close this for now, since it's been open for awhile with no activity. I think my recommendation for this going forward is to use `Timex.now/1`.
